src/features/lightspeed/lightspeedUser.ts

```diff
@@ -300,6 +355,61 @@ export class LightspeedUser {
     return this._userDetails;
   }
 
+  public async getMarkdownLightspeedUserDetails(
+    createIfNone: boolean,
+    useProviderType: AuthProviderType | undefined = undefined,
+  ) {
+    // Ensure we don't try to get a lightspeed auth session when the provider is not initialized
+    if (!this._settingsManager.settings.lightSpeedService.enabled) {
+      return undefined;
+    }
+    if (
+      this._markdownUserDetails &&
+      (!useProviderType || useProviderType === this._userType)
+    ) {
+      return this._markdownUserDetails;
+    }
+
+    await this.setLightspeedUser(createIfNone, useProviderType);
+
+    return this._markdownUserDetails;
+  }
+
+  public async getLightspeedUserContent() {
+    // Ensure we don't try to get a lightspeed auth session when the provider is not initialized
+    if (!this._settingsManager.settings.lightSpeedService.enabled) {
+      return undefined;
+    }
+
+    const markdownUserDetails =
+      await this.getMarkdownLightspeedUserDetails(false);
+    const userDetails = await this.getLightspeedUserDetails(false);
+
+    let content = "";
+    if (markdownUserDetails !== "") {
+      content = String(markdownUserDetails);
+    } else {
+      if (userDetails) {
+        const sessionInfo = getLoggedInUserDetails(userDetails);
+        const userName = userDetails.displayNameWithUserType;
+        const userType = sessionInfo.userInfo?.userType || "";
+        const userRole =
+          sessionInfo.userInfo?.role !== undefined
+            ? sessionInfo.userInfo?.role
+            : "";
+        content = `
+          <p><strong>Logged in as</strong>: ${userName}</p>
+          <p><strong>User Type</strong>: ${userType}</p>
+          ${userRole ? "Role: " + userRole : ""}
+        `;
+      } else {
```

Comment:
I would initialize `content` as `content = "undefined";` and drop this else block.